### PR TITLE
feat: add after_server_boot hook

### PR DESCRIPTION
**Pull request by dflynn15**
# Why?

For a project I'm on, there is a requirement for our GraphQL gateway to be setup _after_ our Rails server is booted. This constraint is due to the gateway setup, but the hook may help with other services or configuration that needs to be stood up alongside the Rails server in test.

### CODE REVIEW

File: lib/cypress-rails/server.rb [hunk 1]
```diff
@@ -78,6 +80,7 @@ def boot
           raise "Rack application timed out during boot" if timer.expired?
 
           @server_thread.join(0.1)
+          @initializer_hooks.run(:after_server_start)
```
Comment (dflynn15) [inline lib/cypress-rails/server.rb:83]: I placed the hook to call here as it appeared to be both the soonest, and arguably most safe place to ensure that the application server has booted.

Comment (searls) [reply lib/cypress-rails/server.rb:83]: seems good

**Comment by searls** [comment]
Is there any way you'll be able to get a test of this by making the example/ app's success depend on having such a hook?

**Comment by dflynn15** [comment]
Added a minimal `puts`, "Good job friend!", example up.

**Comment by searls** [comment]
Thanks! My meaning was to also assert on this from the example app's cypress suite (to ensure that the hook is running and not broken) -- failing that, it'd be good to have some other kind of test to ensure this works

File: example/an_app/config/initializers/cypress_rails_initializer.rb [hunk 2]
```diff
@@ -8,8 +8,11 @@
 end
 
 CypressRails.hooks.after_server_start do
-  # After the server has  booted we add the compliment!
-  puts "Congrats the server has started!"
+  # After the server has  booted we add the compliment to the existing fixture list!
+  Compliment.create(text: "This shall be the first.")
+  if Compliment.count == 4
+    raise "I cannot run tests without compliments!"
```
Comment (searls) [inline example/an_app/config/initializers/cypress_rails_initializer.rb:14]: lol the random model i chose for this example app never fails to make me chuckle. nice one @dflynn15

**Pull request merged**